src/main/java/de/numcodex/sq2cql/model/cql/AgeFunctionInvocation.java

```diff
@@ -0,0 +1,41 @@
+package de.numcodex.sq2cql.model.cql;
+
+import de.numcodex.sq2cql.PrintContext;
+import de.numcodex.sq2cql.model.common.TermCode;
+import java.util.List;
+
+public record AgeFunctionInvocation(String unit) implements InvocationTerm {
+  public static final TermCode AGE = TermCode.of("http://snomed.info/sct", "424144002",
+      "Current chronological age");
+
+
+  public static AgeFunctionInvocation of(String unit) {
+    return new AgeFunctionInvocation(unit);
+  }
+
+  private FunctionInvocation getAgeFunction() {
+    switch (unit) {
+      case "a" -> {
+        return FunctionInvocation.of("AgeInYears", List.of());
+      }
+      case "m" -> {
+        return FunctionInvocation.of("AgeInMonths", List.of());
+      }
+      case "wk" -> {
+        return FunctionInvocation.of("AgeInWeeks", List.of());
+      }
+      case "d" -> {
+        return FunctionInvocation.of("AgeInDays", List.of());
+      }
+      case "h" -> {
+        return FunctionInvocation.of("AgeInHours", List.of());
+      }
+      default -> throw new IllegalArgumentException("Unsupported unit: " + unit);
+    }
+  }
+
+  @Override
+  public String print(PrintContext printContext) {
+    return getAgeFunction().print(printContext);
```

Comment:
Please move that calculation into the `of` method, because there, the IllegalArgumentException makes sense. Here in print, the IllegalArgumentException is not appropriate.